JavaScript/JavaScriptSDK/appInsights.ts

```diff
@@ -301,7 +318,31 @@ module Microsoft.ApplicationInsights {
 
                 this.context.track(envelope);
             } catch (e) {
-                _InternalLogging.warnToConsole("trackTrace failed, trace will not be collected: " + Util.dump(e));
+                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING, "trackTrace failed, trace will not be collected: " + Util.dump(e));
+            }
+        }
+
+        /**
+        * Log a page visit time
+        * @param    pageName    Name of page
+        * @param    pageVisitDuration Duration of visit to the page in milleseconds
+        */
+        public trackPageVisitTime(pageName: string, pageUrl: string, pageVisitTime: number) {
+            var properties = { PageName: pageName, PageUrl: pageUrl };
+            this.trackMetric("PageVisitTime", pageVisitTime, 1, pageVisitTime, pageVisitTime, properties);
+        }
+
+        private autoTrackPageVisitTime(pageName: string, pageUrl: string) {
```

Comment:
I made this change by adding a method: restartTrackPageVisitTime to the PageTimeVisitManager which sends telemetry for the previous page and starts tracking the new page visit.

Let me know if you have ideas of a better method name, but this seems to be consistent with the other start/stop track method names...